src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9953,22 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+int n_get_commands( void )
+{
+  return ucmds.ga_len;
+}
+
+Dictionary get_dict_command(int i) {
+  Dictionary dic = ARRAY_DICT_INIT;
+  ucmd_T *cmd = USER_CMD(i);
+  PUT(dic, "uc_name", STRING_OBJ(cstr_to_string((const char *)cmd->uc_name)));
```

Comment:
these are the internal names, we don't want them in external API. Better to use terms mentioned in documentation `:help command`.